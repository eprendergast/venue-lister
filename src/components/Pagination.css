
.pagination-container {
    display: flex;
    justify-content: center;
  }
  
  .pagination-link-container {
    background-color: var(--secondary-grey);
    color: var(--primary-black);
    margin: 2px;
  }
  
  .pagination-link-container-active {
    background-color: var(--primary-red);
    color: white;
    margin: 2px;
    font-weight: bold;
  }
  
  .pagination-link-container :hover {
    background-color: var(--primary-red);
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  
  .pagination-link-container-active :hover {
    background-color: var(--primary-red);
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  
  .pagination-link {
    padding: 8px 16px;
  }