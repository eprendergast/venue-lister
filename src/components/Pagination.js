import React from 'react'
import './Pagination.css'

const Pagination = ({
  pageNumbers,
  currentPage,
  setCurrentPage,
  nextPage,
  previousPage
}) => {
  return (
    <div className='pagination-container'>
      <div className='pagination-link-container'>
        <div className='pagination-link' onClick={() => previousPage()}>
          Prev
        </div>
      </div>
      {pageNumbers.map(number => (
        <div
          key={number}
          className={
            number === currentPage
              ? 'pagination-link-container-active'
              : 'pagination-link-container'
          }
        >
          <div
            className='pagination-link'
            onClick={() => setCurrentPage(number)}
          >
            {number}
          </div>
        </div>
      ))}
      <div className='pagination-link-container'>
        <div className='pagination-link' onClick={() => nextPage()}>
          Next
        </div>
      </div>
    </div>
  )
}

export default Pagination
