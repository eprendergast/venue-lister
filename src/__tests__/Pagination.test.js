import React from 'react'
import renderer from 'react-test-renderer'
import Pagination from '../components/Pagination.js'

it('renders correctly', () => {
  const pageNumbers = [1, 2, 3, 4, 5]

  const tree = renderer
    .create(<Pagination pageNumbers={pageNumbers} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
