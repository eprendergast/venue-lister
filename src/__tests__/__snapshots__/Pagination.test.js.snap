// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pagination-container"
>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      Prev
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      1
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      2
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      3
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      4
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      5
    </div>
  </div>
  <div
    className="pagination-link-container"
  >
    <div
      className="pagination-link"
      onClick={[Function]}
    >
      Next
    </div>
  </div>
</div>
`;
